crates/acvm/src/lib.rs

```diff
@@ -30,6 +31,7 @@ pub enum GateResolution {
     UnknownError(String),      //Generic error
     UnsupportedOpcode(OPCODE), //Unsupported Opcode
     UnsatisfiedConstrain,      //Gate is not satisfied
+    Solved,                    //Circuit is solved, after a number of passes
```

Comment:
Can we remove the variant entirely? It is only returned by `solve`, but there is no reason `solve` could not return Resolved instead.